app/jobs/notifications/milestone_job.rb

```diff
@@ -0,0 +1,9 @@
+module Notifications
+  class MilestoneJob < ApplicationJob
+    queue_as :send_milestone_notification
+
+    def perform(milestone_hash, service = Notifications::Milestone::Send)
```

Comment:
Here a milestone hash contains an ActiveRecord object (an `article`). I think we should continue the pattern of passing a record `id`, querying the database and then passing the object further if it exists. Though destroying article is unlikely, it still could be the case and could lead to Deserialization errors.

Reply:
I have updated the code to proceed like that.